sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusSessionReceiver.cs

```diff
@@ -147,7 +147,7 @@ public virtual async Task<byte[]> GetSessionStateAsync(CancellationToken cancell
 
             try
             {
-                await InnerReceiver.SetStateAsync(sessionState, cancellationToken).ConfigureAwait(false);
+                await InnerReceiver.SetStateAsync(sessionState.ToBytes().ToArray(), cancellationToken).ConfigureAwait(false);
```

Comment:
Should we move this down so that we pass the BD all the way down?

Reply:
Huh? This is the interface to the AMQP side of things, aka ITransportProtocol. Since we're only changing the public surface, I didn't feel it was necessary to change our internal transport layer to do this. That's a lot more work for no benefit since we won't use it in the lower layers. The only way it would be beneficial is if AMQP worked with ReadOnlyMemory.

Reply:
The benefit is that we would be more explicit and remove dead code that will never be hit.

Reply:
Fixed.